app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -139,7 +143,10 @@ def index_relation
       end
 
       relation = relation.where(Katello::Content.table_name => {:name => params[:name]}) if params[:name].present?
-      relation = relation.redhat unless ::Foreman::Cast.to_bool(params[:with_custom])
+      unless ::Foreman::Cast.to_bool(params[:repository_type])
```

Comment:
This seems more consistent with non boolean params. `params[:repository_type].present?`